### ISSUE TITLE
Inconsistent test result between ansible and ssh backends

### ISSUE THREAD

**Issue by codequokka** [philpep/testinfra#435]
Hi

I just created a simple test to check for python version:
```
def test_python3_is_installed(host):
    c = host.run('/opt/python3.6.4/bin/python3.6 -V')

    assert c.stdout == 'Python 3.6.4'
```

I run pytest against same vagrant instance.
But test results are inconsistent by changing backend:

- Test is passed with ansible backend
```
$ pytest --connection=ansible --ansible-inventory=.vagrant/provisioners/ansible/inventory/vagrant_ansible_inventory --hosts=default -vvv
=================================================================== test session starts ====================================================================
platform darwin -- Python 3.7.3, pytest-4.4.1, py-1.8.0, pluggy-0.9.0 -- /Users/nbchk/.ghq/github.com/codequokka/pypro3-dev-env/.venv/bin/python3.7
cachedir: .pytest_cache
rootdir: /Users/nbchk/.ghq/github.com/codequokka/pypro3-dev-env
plugins: testinfra-2.1.0
collected 1 item

deployment/roles/codequokka.ansible_role_pypro3/molecule/resources/tests/test_python3.py::test_python3_is_installed[ansible://default] PASSED        [100%]

=========================================================== 1 passed in 1.91 seconds ===========================================================
```

- Test is failed with ssh backend
stdout=b'Python 3.6.4\n' has newline
```
$ pytest --connection=ssh --ansible-inventory=.vagrant/provisioners/ansible/inventory/vagrant_ansible_inventory --hosts=default -vvv
=================================================================== test session starts ====================================================================
platform darwin -- Python 3.7.3, pytest-4.4.1, py-1.8.0, pluggy-0.9.0 -- /Users/nbchk/.ghq/github.com/codequokka/pypro3-dev-env/.venv/bin/python3.7
cachedir: .pytest_cache
rootdir: /Users/nbchk/.ghq/github.com/codequokka/pypro3-dev-env
plugins: testinfra-2.1.0
collected 1 item

deployment/roles/codequokka.ansible_role_pypro3/molecule/resources/tests/test_python3.py::test_python3_is_installed[ssh://default] FAILED            [100%]

========================================================================= FAILURES =========================================================================
_________________________________________________________ test_python3_is_installed[ssh://default] _________________________________________________________

host = <testinfra.host.Host object at 0x110aa7320>

    def test_python3_is_installed(host):
        c = host.run('/opt/python3.6.4/bin/python3.6 -V')

>       assert c.stdout == 'Python 3.6.4'
E       AssertionError: assert 'Python 3.6.4\n' == 'Python 3.6.4'
E         - Python 3.6.4
E         ?             -
E         + Python 3.6.4

deployment/roles/codequokka.ansible_role_pypro3/molecule/resources/tests/test_python3.py:33: AssertionError
-------------------------------------------------------------------- Captured log call ---------------------------------------------------------------------
base.py                    268 INFO     RUN CommandResult(command=b"ssh default '/opt/python3.6.4/bin/python3.6 -V'", exit_status=0, stdout=b'Python 3.6.4\n', stderr=None)
================================================================= 1 failed in 0.53 seconds =================================================================
```

Is there any difference between ansible and ssh backend?

- Package info
```
$ pip list
Package        Version
-------------- -------
ansible        2.7.10
asn1crypto     0.24.0
atomicwrites   1.3.0
attrs          19.1.0
bcrypt         3.1.6
cffi           1.12.3
cryptography   2.6.1
Jinja2         2.10.1
MarkupSafe     1.1.1
more-itertools 7.0.0
paramiko       2.4.2
pip            19.1
pluggy         0.9.0
py             1.8.0
pyasn1         0.4.5
pycparser      2.19
PyNaCl         1.3.0
pytest         4.4.1
PyYAML         5.1
setuptools     41.0.1
six            1.12.0
testinfra      2.1.0
wheel          0.33.1
```

**Comment by philpep** [comment]
Hi, thanks for reporting !

I'm not sure if it's a bug or a feature here. Looks ansible "command" module strip the trailing newline here whereas testinfra does not modify `stdout` and `stderr`.

You can use `host.check_output(...)` which will strip the trailing newline here.

Also note that `ansible` backend behavior will change in next testinfra release (which will probably be a 3.0.0), see #431 for details. In particular, this will use testinfra local, ssh or docker backend instead, so you will have the same behavior for both.

**Comment by philpep** [comment]
I just released testinfra 3.0.0 which should fix your issue by side effect ;)

**Comment by codequokka** [comment]
Thanks for replying.

After upgrading testinfra to 3.0.0, stdout has trailing newline in both ansible and ssh backends.
Thank you for bug fixing!
